symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java

```diff
@@ -1,53 +1,79 @@
         assertEquals(config.getBot().getUsername(), "tibot");
     }
 
     @Test
-    public void loadFromYamlFileTest() throws JsonProcessingException, BdkConfigException {
-        String configPath = System.getProperty("user.dir") + "/src/test/resources/config.yaml";
+    public void loadFromYamlFileTest() throws BdkConfigException {
+        String configPath = System.getProperty("user.dir") + "/src/test/resources/config/config.yaml";
         BdkConfig config = BdkConfigLoader.loadFromFile(configPath);
         assertEquals(config.getBot().getUsername(), "tibot");
     }
 
     @Test
-    public void loadFromJsonFileTest() throws JsonProcessingException, BdkConfigException {
-        String configPath = System.getProperty("user.dir") + "/src/test/resources/config.json";
+    public void loadFromJsonFileTest() throws BdkConfigException {
+        String configPath = System.getProperty("user.dir") + "/src/test/resources/config/config.json";
         BdkConfig config = BdkConfigLoader.loadFromFile(configPath);
         assertEquals(config.getBot().getUsername(), "tibot");
     }
 
     @Test
-    public void loadFromJsonClasspathTest() throws JsonProcessingException, BdkConfigException {
-        BdkConfig config = BdkConfigLoader.loadFromClasspath("/config.json");
+    public void loadFromJsonClasspathTest() throws BdkConfigException {
+        BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config.json");
         assertEquals(config.getBot().getUsername(), "tibot");
     }
 
     @Test
-    public void loadFromYamlClasspathTest() throws JsonProcessingException, BdkConfigException {
-        BdkConfig config = BdkConfigLoader.loadFromClasspath("/config.yaml");
+    public void loadFromYamlClasspathTest() throws BdkConfigException {
+        BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config.yaml");
         assertEquals(config.getBot().getUsername(), "tibot");
     }
+
+    @Test
+    public void loadFromFileNotFoundTest() throws BdkConfigException {
+        BdkConfigException exception = assertThrows(BdkConfigException.class, () -> {
+            String configPath = System.getProperty("user.dir") + "/wrong_path/config.yaml";
+            BdkConfigLoader.loadFromFile(configPath);
+        });
+        assertEquals(exception.getMessage(), "Config file is not found");
+    }
+
+    @Test
+    public void loadLegacyFromInputStreamTest() throws BdkConfigException {
+        InputStream inputStream = BdkConfigLoaderTest.class.getResourceAsStream("/config/legacy_config.json");
+        BdkConfig config = BdkConfigLoader.loadFromInputStream(inputStream);
+        assertEquals(config.getBot().getUsername(), "tibot");
+        assertEquals(config.getBot().getPrivateKeyPath(), "/Users/thibault.pensec/local/conf/agent/privatekey.pem");
+        assertEquals(config.getApp().getPrivateKeyPath(), "/Users/thibault.pensec/local/conf/agent/privatekey.pem");
```

Comment:
Please remove "thibault.pensec" from the test data 😅  I'll do the same with my PR !